app/controllers/advertisements_controller.rb

```diff
@@ -0,0 +1,69 @@
+class AdvertisementsController < ApplicationController
+  before_action :set_advertisement, only: [:show, :edit, :update, :destroy]
+  before_action :not_found_unless_author, only: [:edit, :update, :destroy]
+  PER_PAGE=10
+
+  def index
+    @advertisements = Advertisement.
+      displayable.
+      page(params[:page]).per(PER_PAGE)
+  end
+
+  def all
+    @advertisements = Advertisement.page(params[:page]).per(PER_PAGE)
+    render "advertisements/index"
+  end
+
+  def show
+  end
+
+  def new
+    @advertisement = Advertisement.new
+  end
+
+  def edit
+  end
+
+  def create
+    @advertisement = Advertisement.new(advertisement_params)
+    @advertisement.user = current_user
+
+    if @advertisement.save
+      redirect_to @advertisement,
+        notice: 'Advertisement was successfully created.'
+    else
+      render :new
+    end
+  end
+
+  def update
+    if @advertisement.update(advertisement_params)
+      redirect_to @advertisement,
+        management: true,
+        notice: 'Advertisement was successfully updated.'
+    else
+      render :edit
+    end
+  end
+
+  def destroy
+    @advertisement.destroy
+
+    redirect_to all_advertisements_path, notice: 'Advertisement was successfully destroyed.'
```

Comment:
Line is too long. [92/80]<br>Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.